scripts/scil_compute_ihMT_maps.py

```diff
@@ -0,0 +1,320 @@
+    ihMTsat = (1/T1appb)-(1/T1appa)
+
+    return ihMTR, ihMTsat
+
+
+def compute_MT_maps(contrast_maps, acq_parameters):
+    """
+    Compute MT maps : description details coming soon !
+
+    Parameters
+    ----------
+    contrast_maps:      List of constrats matrices : list of all contrast maps
+                        computed with compute_ihMT_contrasts
+    acq_parameters:     Lists of RT and Flipangle for ihMT and T1w
+                        [RT, Flipangle]
+    Returns
+    -------
+    Non-ihMT maps in 3D-Array : MT ratio and saturation Maps.
+    """
+    # Compute MT Ratio map
+    MTR = 100*((contrast_maps[2]-(contrast_maps[4]+contrast_maps[3])/2)/contrast_maps[2])
+
+# flake8 to fix: meme rangaine longueur ligne où couper ???
+    # Compute MT sat maps
+    cPD1 = contrast_maps[2]
+    cPD2 = (contrast_maps[4]+contrast_maps[3])/2
+    cT1 = contrast_maps[5]
+    Aapp = ((2*acq_parameters[0][0]/(acq_parameters[0][1]**2))-(2*acq_parameters[1][0]/(acq_parameters[1][1]**2)))/(((2*acq_parameters[0][0])/(acq_parameters[0][1]*cPD1))-((2*acq_parameters[1][0])/(acq_parameters[1][1]*cT1)))
+    T1app = ((cPD1/acq_parameters[0][1])-(cT1/acq_parameters[1][1]))/((cT1*acq_parameters[1][1])/(2*acq_parameters[1][0])-(cPD1*acq_parameters[0][1])/(2*acq_parameters[0][0]))
+    MTsat = 100*(((Aapp*acq_parameters[0][1]*acq_parameters[0][0]/T1app)/cPD2)-(acq_parameters[0][0]/T1app)-(acq_parameters[0][1]**2)/2)
+
+    return MTR, MTsat
+
+
+def threshold_ihMT_maps(computed_map, contrast_maps, in_mask, lower_threshold,
+                        upper_threshold, idx_contrast_list):
+    """
+    Remove NaN, apply different threshold based on
+       - maximum and minimum threshold value
+       - combination of specific contrasts maps
+    Multiple data by probability T1 mask (GM+WM+CSF)
+
+    Parameters
+    ----------
+    computed_map        3D-Array data.
+                        Myelin map (ihMT or non-ihMT computed maps)
+    contrast_maps       List of 3D-Array. File must containing the
+                        5 contrasts maps.
+    in_mask             Path to probability T1 map
+    lower_threshold     Value for low thresold <int>
+    upper_thresold      Value for up thresold <int>
+    idx_contrast_list   List of indice contrasts maps in order corresponding to
+                        contrast_maps input ex.: [1, 2, 3]
+                        Altnp = 0; Atlpn = 1; Reference = 2; Negative = 3;
+                        Positive = 4; T1weighted = 5
+    Returns
+    ----------
+    Thresholded matrice in 3D-array.
+    """
+    # Remove NaN and apply thresold based on lower and upper value
+    computed_map[np.isnan(computed_map)] = 0
+    computed_map[np.isinf(computed_map)] = 0
+    computed_map[computed_map < lower_threshold] = 0
+    computed_map[computed_map > upper_threshold] = 0
+
+    # Load and apply probability masks on maps
+    mask_image = nib.load(in_mask)
+    mask_data = mask_image.get_fdata(dtype=np.float32)
```

Comment:
if it's a mask it should be loaded as get_data_as_mask. I realise your mask is a little bit weird to me you should ask for a binary mask even if you describe it as sum of GM WM CSF.